64_allowlist_signing.md

```diff
@@ -0,0 +1,290 @@
+- User provides the tenant with paths to an allowlist, a signature, and a public key using a modern algorithm (e.g. ECDSA)
+- Keylime verifies the inputs are consistent, then checks Rekor for an inclusion proof of those artifacts, before loading them into Keylime
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+This proposal suggests the [Sigstore](https://www.sigstore.dev/) project for use as a transparency log.
+
+Key terminology: “Rekor” is the transparency log Sigstore uses to store artifacts.
+
+### Allowlist loading
+
+Currently, all allowlist verification happens on the tenant when loading an allowlist file from disk. If the user specifies a signature file and key file using the `--allowlist-sig` and `--allowlist-sig-key` files, the tenant will first verify that the signature matches the provided allowlist file using the public key.
+
+Two fundamental changes/additions should be made to this process.
+
+#### More key types
+
+Right now, the tenant assumes that all signing is done using GPG keys. Instead, the tenant should inspect the provided key file to determine which algorithm is being used (for example, it could be an ECDSA key) and behave accordingly. This will allow for future extensibility around new key algorithms, and will not require the user to specify what kind of key they are providing.
+
+#### Optional verification step against a transparency log
+
+Currently, all signature validation happens within `read_allowlist()` in [ima.py](https://github.com/keylime/keylime/blob/master/keylime/ima.py#L408). This proposal suggests an additional, optional verification step that would check provided artifacts for inclusion proof in a transparency log. This assumes the user has already uploaded those artifacts to that transparency log using a separate tool (for example, `rekor-cli`).
+
+To support this new functionality, the tenant command should have two new flags:
+
+- `--allowlist-verify-transparency-log`: to indicate the user would like to verify against a transparency log. Since both a signature and public key are necessary, existing flags`--allowlist-sig` and `--allowlist-sig-key` should also be set when using this new flag.
```

Comment:
just a nit, but i would go for `allowlist-verify-tlog` or `rekor-verify` to save a few keystrokes for users :)

Reply:
Agreed. We're already using some pretty long options to the tenant, no need to make it worse.